### PR TITLE
Simplify ld-script

### PR DESCRIPTION
**Pull request by AtariDreams**
We should just wildcard as much as we can where order does not matter. It would help, I think.

I also put decompress and malloc together since the buffers add up nicely to 0x20000.

## **Discord contact info**
roseareazero

### CODE REVIEW

File: ld_script_modern.ld [hunk 1]
```diff
@@ -15,26 +15,20 @@ SECTIONS {
     ewram 0x2000000 (NOLOAD) :
     ALIGN(4)
     {
-        src/*.o(ewram_data);
-        gflib/*.o(ewram_data);
+        gflib/malloc.o(ewram_data);
+        src/decompress.o(ewram_data);
+        . = 0x20000;
```
Comment (mrgriffin) [inline ld_script_modern.ld:20]: Is this necessary? If we just had `*.o(ewram_data);` would anything break as a result of the heap/decompression buffer having moved?

File: ld_script_modern.ld [hunk 2]
```diff
@@ -15,26 +15,20 @@ SECTIONS {
...
-        *libc.a:*.o(.bss*);
-        *libnosys.a:*.o(.bss*);
-
-        src/m4a.o(.bss.code);
-
-        src/*.o(COMMON);
-        gflib/*.o(COMMON);
-        *libc.a:*.o(COMMON);
-        *libnosys.a:*.o(COMMON);
+        *(.dynbss)
```
Comment (mrgriffin) [inline ld_script_modern.ld:28]: What's in the `.dynbss` section?

File: ld_script_modern.ld [hunk 3]
```diff
@@ -83,9 +77,7 @@ SECTIONS {
     .rodata :
     ALIGN(4)
     {
-        src/*.o(.rodata*);
-        gflib/*.o(.rodata*);
-        data/*.o(.rodata*);
+        *(.rodata*)
```
Comment (mrgriffin) [inline ld_script_modern.ld:80]: How come you've written this as `*(.rodata*)` instead of `*(.rodata)` and `*(.rodata.*)` like you did with `.text`/`.bss`?

**Review by mrgriffin — COMMENTED** [review]
How come some of the linker script lines end in a `;`, but others don't? I think it would be worth being consistent one way or the other?

File: ld_script_modern.ld (hunk 1 above)
Comment (AtariDreams) [reply ld_script_modern.ld:20]: It has to be word aligned for starters so this helps

**Comment by AtariDreams** [comment]
> How come some of the linker script lines end in a `;`, but others don't? I think it would be worth being consistent one way or the other?

Fixed!

**Comment by Bassoonian** [comment]
Could you address MGriffin's other concerns? If not, this doesn't qualify for a merge. Thank you!

**Comment by Bassoonian** [comment]
Closing this for the time being, feel free to re-open later

**Pull request closed without merging**